build/patches/Bromite-AdBlockUpdaterService.patch

```diff
@@ -364,18 +472,109 @@ new file mode 100644
++        mAdBlockEdit.setSummary(AdblockUpdaterBridge.getAdBlockFiltersURL());
++
++        DateFormat df = DateFormat.getDateTimeInstance(DateFormat.MEDIUM, DateFormat.MEDIUM);
++        currentIndexVersion.setSummary(
++            AdblockUpdaterBridge.getAdBlockMostRecentIndexedVersion() +
++            "\n" + getString(R.string.adblock_last_checked_text) +
++            " "  + df.format(AdblockUpdaterBridge.getAdBlockLastUpdate()));
 +    }
 +
-+    private void updateCurrentAdBlockUrl() {
-+        mAdBlockEdit.setSummary(CachedFeatureFlags.getAdBlockFiltersURL());
++    public static String GetAdBlockMessage(@AdblockEvent int event, @AdblockError int error) {
++        Context applicationContext = ContextUtils.getApplicationContext();
++        String message = "";
++        if (error == AdblockError.NONE) {
++            if (event == AdblockEvent.ADBLOCK_CHECKING_FOR_UPDATES) message = applicationContext.getString(R.string.options_adblock_event_1);
++            else if (event == AdblockEvent.ADBLOCK_UPDATE_DOWNLOADING) message = applicationContext.getString(R.string.options_adblock_event_3);
++            else if (event == AdblockEvent.ADBLOCK_UPDATED) message = applicationContext.getString(R.string.options_adblock_event_4);
++        }
++        else if (error == AdblockError.UPDATE_NOT_NEEDED) message = applicationContext.getString(R.string.options_adblock_error_3);
++        else if (error == AdblockError.DOWNLOAD_ERROR) message = applicationContext.getString(R.string.options_adblock_error_4);
++        return message;
++    }
++
++    public void onAdBlockUpdaterResult(@AdblockEvent int event, @AdblockError int error) {
++        String message = GetAdBlockMessage(event, error);
++        TextMessagePreference currentStatus = (TextMessagePreference) findPreference(PREF_ADBLOCK_CURRENT_STATUS);
++        currentStatus.setTitle(message);
++
++        updateCurrentAdBlockSettings();
 +    }
 +
 +    @Override
 +    public void onResume() {
 +        super.onResume();
-+        updateCurrentAdBlockUrl();
++        updateCurrentAdBlockSettings();
++    }
++
++    private TimeFrequencySpinnerOption[] getTimeFrequencySpinnerOptions() {
++        List<TimeFrequencySpinnerOption> options = new ArrayList<>();
++        options.add(new TimeFrequencySpinnerOption(1,
++                getString(R.string.options_adblock_frequency_day)));
++        options.add(new TimeFrequencySpinnerOption(4,
++                getString(R.string.options_adblock_frequency_days)));
++        options.add(new TimeFrequencySpinnerOption(7,
++                getString(R.string.options_adblock_frequency_days)));
++        options.add(new TimeFrequencySpinnerOption(10,
++                getString(R.string.options_adblock_frequency_days)));
++        options.add(new TimeFrequencySpinnerOption(30,
++                getString(R.string.options_adblock_frequency_days)));
++        return options.toArray(new TimeFrequencySpinnerOption[0]);
```

Comment:
I would suggest these options:
* Never
* 7 days (default)
* 15 days
* 30 days

In fact the filters are updated every 2 weeks, that is why the default is 7 days.

Reply:
fixed in https://github.com/bromite/bromite/pull/905/commits/80c163754a3afa025bd7004aec20d43df53a4c12